src/Host/Protocol/Impl/CustomHttpError.cs

```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.R.Host.Protocol {
+    public enum CustomHttpError {
```

Comment:
We should not be using custom HTTP error codes - this is contrary to REST. We should definitely not be using codes from 5xx range, these are for _unexpected_ server errors. And 4xx range, which is the right one here, is too crowded already, and does not permit custom code assignment anyway.

In most cases, we should be reporting HTTP error 400 for cases like these, and add our own error code somewhere (either a custom header, or as part of error message JSON).